src/tool_formparse.c

```diff
@@ -381,8 +384,18 @@ static char *get_param_word(char **str, char **end_pos, char endchar)
           while(ptr < *end_pos);
           *end_pos = ptr2;
         }
-        while(*ptr && *ptr != ';' && *ptr != endchar)
+        ++ptr;
+        while(*ptr && *ptr != ';' && *ptr != endchar) {
+          if(!ISSPACE(*ptr))
+            trailing_data = TRUE;
           ++ptr;
+        }
+        if(trailing_data) {
+          char oldchr = *ptr;
+          *ptr = '\0';
+          warnf(config->global, "Ignoring trailing data: %s\n", trail_start);
+          *ptr = oldchr;
+        }
```

Comment:
I wonder if there is enough value in printing the actual trailing data, or if we should simplify and just warn that there was data at all?

Reply:
I figured that printing the data can be a lot clearer than explaining the abstract rule, particularly if you have a longer command and you'd have to figure out which part it's coming from.